### DISCUSSION TITLE
How to use Tetra outside Django Models ?

### DISCUSSION THREAD

**Discussion by bitplorer** [tetra-framework/tetra#7] [Ideas]
First of all Tetra looks Great!. :)

Can we use Tetra with EdgeDB models as well. Can we keep the model logic entirely separate from Django Models so that we can use Tetra as we wish to.

Also I have developed a library (still private) entirely written  in python where we can create Custom Elements and Web Components purely from python. for example as follows:

```python



@dataclass
class CustomElement(TemplateElement):

    def __render__(self, tag_name):
        return template(img(x_component=tag_name))

# now we can declare custom element  
image = CustomElement(tag_name="img")

# now we may use it like
image(src="http://....",  className="h-32 w-32", alt="some image").render() 

# this spits out html string 
<x-img alt="some image" class="h-32 w-32" src="http://...."></x-img> 

```
I have already developed partial support for Websockets on these custom elements. I don't know how to push data state updates though :)

Edit: It also depends heavily on alpinejs

**Comment by samwillis** [comment]
Thanks,

Tetra components are not tied to Django models so you can use any persistence/db/model layer you want. On my to do list is a `ModelComponent` which will bind to a Django Model. But yes, you can use EdgeDB with Django.

Websockets (or SSE) are planned for Tetra, still finalising plans.

**Reply by bitplorer** [reply]
Sure, thanks for the input :). 

I would like to suggest you one thing for supporting inlined CSS in ".py" file via tailwindcss . Just install tailwindcss standalone cli and point your .py files inside tailwind.config.js, something like:
`module.exports = {
    mode: "jit",
    content: {
        files:[
        "../ecommerce/site/**/*.{html,js,jsx,py,vue}",
        "../ecommerce/site/**/**/*.{html,js,jsx,py,vue}",
        ]
      },
}
`

see their is a **'py'** extension inside 'files' list.
Now tailwindcss will pick all your css classes inside .py files :+1:

**Reply by samwillis** [reply]
Thanks, thats super helpful.
